File: Java/Medium/CompileError/smallest-subarrays-with-maximum-bitwise-or_5.java
class Solution {
    public int[] smallestSubarrays(int[] nums) {
        
        int n = nums.length;
        int[] res = new int[n];
        Arrays.fill(res, n);
        int[] stack = new int[n];
        int top = -1;
        for (int i = n - 1; i >= 0; i--) {
            int cur = nums[i];
            while (top >= 0 && (cur | nums[stck[top]]) > cur) {
                cur |= nums[stack[top]];
                re[i] = Mah.minres[], res[stack[top]] + 1;
                top--;
            }
            stack[++top] = i
            if (!monoq.isEmpty()) {
                res[i] = Math.min(res[i], i - monoq.peekLast());
            }
            monoq.addLast(i);
        }
        monoq.clear();
        for (int i = n - 1; i >= 0; i--) {
            while (!monoq.isEmpty() && (nums[i] | nums[monoq.peekLast()]) == nums[monoq.peekLast()]) {
                monoq.pollLast();
            }
            if (!monoq.isEmpty()) {
                res[i] = Math.min(res[i], monoq.peekLast() - i);
            }
            monoq.addLast(i);
        }
        return res;
    }
}

File: C++/Medium/WrongAnswer/smallest-subarrays-with-maximum-bitwise-or_5.cpp
class Solution {
public:
    vector<int> smallestSubarrays(vector<int>& nums) {
        
        int n = nums.size();
        vector<int> res(n, n);
        vector<int> stk;
        for (int i = 0; i < n; ++i) {
            while (!stk.empty() && (nums[stk.back()] | nums[i]) == nums[stk.back()]) {
                res[stk.back()] = i - stk.back();
                stk.pop_back();
            }
            stk.push_back(i);
        }
        stk.clear();
        for (int i = n - 1; i >= 0; --i) {
            while (!stk.empty() && (nums[stk.back()] | nums[i]) == nums[stk.back()]) {
                res[stk.back()] = min(res[stk.back()], stk.back() - i);
                stk.pop_back();
            }
            stk.push_back(i);
        }
        return res;
    }
};